eos-enable-zram

```diff
@@ -13,13 +13,79 @@ if [ "$1" -eq 0 ]; then
     exit 0
 fi
 
-ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
+# List and disable any active swap areas
+active_swaps=$(swapon --show=NAME --raw --noheadings)
+echo "Active swap areas before $(basename $0): ${active_swaps}"
+swapoff -a
 
+# Enable zram
+ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
 if [ ! -e /sys/block/zram0 ]; then
     modprobe zram
 fi
 echo $(($ram_size_kb * 2))K > /sys/block/zram0/disksize
-# For now, disable the swap partition if in use
-swapoff -a
 mkswap /dev/zram0
 swapon /dev/zram0
+
+if [ -z "${active_swaps}" ]; then
+    exit 0
+fi
+
+# Find root partition, root partition number, root disk, and table type
+orig_root_part=$(systemctl show -p What sysroot.mount)
+orig_root_part=${orig_root_part#What=}
+root_part=$(readlink -f "${orig_root_part}")
+case ${root_part} in
+    /dev/sd??)
+        root_disk=${root_part%?}
+        ;;
+    /dev/*p[0-9])
+        root_disk=${root_part%p?}
+        ;;
+esac
+
+case ${orig_root_part} in
+    /dev/mapper/endless-image?)
+        root_disk=${orig_root_part%?}
+        ;;
+esac
+
+if [ -z "${root_disk}" ]; then
+    exit 0
```

Comment:
Shouldn't this be a non-zero exit?

Reply:
The main purpose of this service is to enable-zram, which at this point happened successfully, so I chose to treat reclaiming the swap space as an extra, and not exit with an error if we fail to do it. Maybe this is a strong enough argument to have this a separate service (which will need to be fully done before eos-enable-zram since I'm using the active swap areas as an input)?

Reply:
I don't think having a separate service would really change things. This service is only WantedBy=multi-user.target as opposed to RequiredBy so it failing would not break the boot process; and whether the `exit 1` is in this service or another one, there'd still just be one failing service, right?

My concern is that this makes it harder to verify that this script is working correctly, whether in a (hypothetical) test suite or in manual testing on installed systems.

Reply:
I don't think failing the service would affect the boot process at all, it is just that in the current situation it would indicate the "enable zram" service had failed when we actually had zram up and running (which could lead to confusion during other analysis).

But I hear you, so IMO the best approach is to separate this code in a separate service (which I was trying to avoid, but this grew more than I initially expected that it actually is a whole separate thing we are doing).